src/mscorlib/src/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -1872,6 +1900,38 @@ unsafe public static IntPtr StringToCoTaskMemUni(String s)
         }
 
         [System.Security.SecurityCritical]  // auto-generated_required
+        unsafe public static IntPtr StringToAllocatedMemoryUTF8(String s)
+        {
+            const int MAX_UTF8_CHAR_SIZE = 3;
+            if (s == null)
+            {
+                return IntPtr.Zero;
+            }
+            else
+            {
+                int nb = (s.Length + 1) * MAX_UTF8_CHAR_SIZE;
```

Comment:
We are probably allocating more than required here , we can allocate the correct size using GetByteCount